### ISSUE TITLE
Github hook integration

### ISSUE THREAD

**Issue by willardmr** [willardmr/sustainable_infrastructure_ci#6]
Someone should be able to easily integrate sustainable CI into their github hooks.

**Comment by AntoineSebert** [comment]
List of github hooks available, with an explanation for the ones that seem not to be relevant:
- **`pre-commit`**
- `prepare-commit-msg` / `commit-msg`: no need to mess with commit message
- **`post-commit`**
- `applypatch-msg` / `pre-applypatch` / `post-applypatch`: email workflows, not very relevant
- `pre-rebase` / `post-rewrite` / `post-checkout`: no new code *per se*
- **`post-merge`**
- **`pre-push`**
- `pre-auto-gc`: no need during garbage collection
- `pre-receive` / `update` / `post-receive`: at this we've lost the benefits of hooks for our use case

I suggest we provide the hooks `pre-commit` / `post-commit` / `post-merge` / `pre-push` by default, with documentation. The other can be derived from them, would they ever be necessary.

Source: https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks

**Comment by AntoineSebert** [comment]
https://docs.github.com/en/actions/using-workflows/reusing-workflows
I think this might be what you have in mind ?
Essentially we just make a workflow with input and outputs such that other workflows can call it. This might be the first step to have it listed on the marketplace.

**Comment by willardmr** [comment]
Yeah exactly!  I have used those in the past and they work really well.  We may want to dockerize our tool since a lot of other workflows seem to do that, but besides that seems like the easiest and most useful thing.  What do you think?
